TorSharp/Tools/LineByLineConfigurer.cs

```diff
@@ -57,14 +57,24 @@ public LineByLineConfigurer(IConfigurationDictionary configurationDictionary, IC
                     }
                 }
 
+                // If the original file exists, remove it before copying over the temporary file.
                 if (File.Exists(path))
                 {
-                    File.Replace(temporaryPath, path, null);
-                }
-                else
-                {
-                    File.Move(temporaryPath, path);
+                    string backupPath = path + ".bak";
+
+                    // If there has already been a backup, just delete the original.
+                    if (File.Exists(backupPath))
+                    {
+                        File.Delete(path);
```

Comment:
Totally fine to have an alternative to `File.Replace`. In an ideal world, a repro for why `Replace` breaks would be good, but file concurrency can be Hard.

> Let me know know, and I'll amend the backup to be:

Yes, let's go with that.